src/proxy/src/grpc/prom_query.rs

```diff
@@ -81,6 +81,21 @@ impl Proxy {
             msg: "Missing context",
             code: StatusCode::BAD_REQUEST,
         })?;
+
+        // Check if the tenant is authorized to access the database.
+        if !self
```

Comment:
Use interceptor to do auth check for gRPC
https://github.com/hyperium/tonic/blob/master/examples/src/interceptor/server.rs

For HTTP, see https://stackoverflow.com/questions/54988438/how-to-check-the-authorization-header-using-warp